tests/raft_scenarios/suffix_collision

```diff
@@ -0,0 +1,70 @@
+nodes,0,1,2
+connect,0,1
+connect,1,2
+connect,0,2
+
+# Node 0 starts first, and begins sending messages first
+periodic_one,0,110
+dispatch_all
+
+periodic_all,10
+dispatch_all
+
+state_all
+
+replicate,1,hello world
+periodic_all,10
+dispatch_all
+
+replicate,1,saluton mondo
+periodic_all,10
+dispatch_all
+
+periodic_all,10
+dispatch_all
+
+state_all
+
+# Node 0 is partitioned, and replicates another item
+disconnect,0,1
+disconnect,0,2
+
+replicate,1,drop me 1
+periodic_all,10
+dispatch_all
+
+state_all
+
+# Node 1 calls an election, and wins
+periodic_one,1,110
+dispatch_all
+
+periodic_all,10
+dispatch_all  #< This AppendEntries starts at a point that Node 1 would accept, but is dropped
+
+state_all
+
+# Node 1 replicates a new entry, resulting in the same seqno as Node 0
+replicate,2,keep me 1
+periodic_all,10
+dispatch_all
+
+replicate,2,keep me 2
+periodic_all,10
+dispatch_all
+
+state_all
+
+# Network heals
+connect,0,1
+connect,0,2
+
+# Node 0 is brought up-to-date
+periodic_all,10
+dispatch_all
+
+periodic_all,10
+dispatch_all
```

Comment:
For the sake of completeness, here's the generated Mermaid traces showing the situation we were stuck in before #2835:

Node 0 hears about the new term and rolls back, but NACKs with a higher-than-matching index
![image](https://user-images.githubusercontent.com/6000239/126758721-cf951cce-9c3e-48ef-af4d-37dbd6a5d6da.png)

Node 1 keeps sending AppendEntries that start too late, and Node 0 (correctly) keeps refusing them
![image](https://user-images.githubusercontent.com/6000239/126758735-2bdf529d-4401-443b-8844-b6e804cdd3c5.png)


Now that #2835 is in, we NACK with a matching index
![image](https://user-images.githubusercontent.com/6000239/126758880-aa914175-6541-425c-ab73-2dac7b0e79be.png)

And as a result we get an AppendEntries starting from `1.2` rather than `2.3`
![image](https://user-images.githubusercontent.com/6000239/126758932-8df10de8-13b8-4bbf-85ad-5cbecd051c70.png)